### ISSUE TITLE
Change base version of CMake

### ISSUE THREAD

**Issue by russel** [Zipios/Zipios#17]
Currently the build allows for CMake as old as 2.8, I wonder if now is the time to raise the floor version of CMake to say 3.2 (which is the version on Travis-CI). This will allow a lot of simplification of the CMake files I believe.

**Comment by AlexisWilke** [comment]
Yeah, I suppose we could at least have version 3. I do not know the exact differences between the various version. On Ubuntu 16.04 I have 3.5.1. Although if it still works with 2.8, there is probably no need to change that, unless you want to use new features in 3 while changing something.
